packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -108,6 +109,15 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
         throw Exception('NetworkImage is an empty file: $resolved');
 
       return decode(bytes);
+    } catch (e) {
+      // Depending on where the exception was thrown, the image cache may not
+      // have had a chance to track the key in the cache at all.
+      // Schedule a microtask to give the cache a chance to add the key.
+      scheduleMicrotask(() {
+        final bool removed = PaintingBinding.instance.imageCache.evict(key);
+        assert(removed);
```

Comment:
This assert may fail if some other code evicts the entry from the image cache before the microtask runs.

For example, if another part of an app does a manual eviction on the network image in a microtask scheduled before this one.

Reply:
My first thought was that that would never be a valid thing to do - I still think that would be pretty unusual - but we have another test that happens to exercise the provider without using the cache at all, which is also definitely valid, and this assert fails there.  Removed.